index.rst

```diff
@@ -1,51 +1,387 @@
+  Once the convergence metric changes by less than a specified level between iterations we can safely exit the loop since additional iterations will provide insignificant improvement.
+  This is slower since templates must be created for each image, for every iteration, but that has the advantage of allowing convergence to be checked for each individual image at no extra cost (see "Dynamic weights" above).
+  If the extra computational cost is not considered prohibitive, then the second test of convergence is far superior, since it is more accurate and enables additional tests and weighting.
+
+* Test for divergence.
+  If a template is made for testing convergence, we should naturally test also for divergence.
+  If the convergence metric actually degrades with a new iteration, that is a clear sign that some feature of the image is being modeled incorrectly and more iterations will only exacerbate the problem.
+  One special case is if only some of the images degrade, because it is possible that they contain astrometric or other errors, and we could choose to continue with those observations flagged (see "Dynamic weights" above).
+  Otherwise, it is safest to exit immediately and discard the solution from the current iteration, using the last solution from before it started to diverge instead.
+
+  * One possible modification is to calculate a spatially-varying convergence metric, and mask regions that degrade in future iterations.
+    This allows an improved solution to be found even if one very bright feature (such as an improperly-masked saturated star or cosmic ray) is modeled incorrectly.
+
+
+Note that if a convergence test is used, it should only be allowed to exit the loop after a minimum number of iterations have passed.
+It will depend on how the convergence metric is calculated and the choice of initial solution, but the first iteration can show a slight degradation of convergence.
+
+
+Examples with simulated images
+------------------------------
+
+To test the above algorithm, I first ran it on images simulated using `StarFast <http://dmtn-012.lsst.io/en/latest/>`_.
+As shown in :numref:`fig-sim_image` below, these images contain a moderately crowded field of stars (no galaxies) with Kolmogorov double-gaussian PSFs, realistic SEDs, photon shot noise, and no other effects other than DCR.
+In this example, the model is built using three frequency planes and eight input simulations of the field, with airmass ranging between 1.0 and 2.0 (*not* including the simulated observation shown in :numref:`fig-sim_image`).
+I use :eq:`eqn-basic_template` to build a DCR-matched template for the simulated observation (:numref:`fig-sim_template`), and subtract this template to make the difference image (:numref:`fig-sim_template_diff`).
+For comparison, in :numref:`fig-sim_image_diff` I subtract a second simulated image generated with the same field 10 degrees closer to zenith.
```

Comment:
You should definitely say more about these diffs.  E.g. what are the residuals in Figure 6 (and what are their signal to noise).  The hamburgers in Figure 7 look strange to people that do this.  I know that PSF modeling will change how things look, but I think those residuals will worry people coming to this for the first time.

Reply:
I echo Simon here--I think you want to emphasize that these are direct pixel subtractions and not Alard & Lupton-style image differences.